sdk/core/System.ClientModel/src/Pipeline/ClientLoggingPolicy.cs

```diff
@@ -0,0 +1,519 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// TODO.
+/// </summary>
+public class ClientLoggingPolicy : PipelinePolicy
+{
+    private const double RequestTooLongTime = 3.0; // sec
+    private readonly ClientModelEventSource s_eventSource = ClientModelEventSource.Singleton("System.ClientModel");
+
+    private readonly bool _logContent;
+    private readonly int _maxLength;
+    private readonly string? _assemblyName;
+    private readonly string? _clientRequestIdHeaderName;
+    private readonly bool _isLoggingEnabled;
+
+    /// <summary>
+    /// TODO.
+    /// </summary>
+    /// <param name="isLoggingEnabled"></param>
+    /// <param name="loggedHeaderNames"></param>
+    /// <param name="loggedQueryParameters"></param>
+    /// <param name="requestIdHeaderName"></param>
+    /// <param name="logContent"></param>
+    /// <param name="maxLength"></param>
+    /// <param name="assemblyName"></param>
+    public ClientLoggingPolicy(
+        bool isLoggingEnabled = true,
+        List<string>? loggedHeaderNames = default,
+        List<string>? loggedQueryParameters = default,
+        string? requestIdHeaderName = default,
+        bool logContent = false,
+        int maxLength = 4 * 1024,
```

Comment:
It'd be good to define constants like this in private const/static variables like we did for ClientRetryPolicy with e.g. DefaultMaxRetries, DefaultInitialDelay, etc.